NightscoutServiceKit/NightscoutService.swift

```diff
@@ -375,17 +368,149 @@ extension NightscoutService: RemoteDataService {
+    }
+    
+    private func commandSource(notification: [String: AnyObject]) throws -> RemoteCommandSource {
         
-        var errorDescription: String? {
-            switch self {
-            case .missingOTP:
-                return "Error: Password is required."
-            }
+        guard let versionString = notification["version"] as? String else {
+            return commandSourceV1 //fallback before versions were introduced
+        }
+        
+        guard let version = Double(versionString) else {
+            throw NightscoutServiceError.missingCommandSource
+        }
+        
+        switch version {
+        case 1.0..<2.0:
+            return commandSourceV1
+        case 2.0..<3.0:
+            return commandSourceV2
+        default:
+            throw NightscoutServiceError.missingCommandSource
+        }
+    }
+
+}
+
+extension NightscoutService: RemoteCommandSourceV1Delegate {
+    
+    func commandSourceV1(_: RemoteCommandSourceV1, handleAction action: Action) async throws {
+        
+        switch action {
+        case .temporaryScheduleOverride(let overrideCommand):
+            try await self.serviceDelegate?.handleRemoteOverride(
+                name: overrideCommand.name,
+                durationTime: overrideCommand.durationTime,
+                remoteAddress: overrideCommand.remoteAddress
+            )
+        case .cancelTemporaryOverride:
+            try await self.serviceDelegate?.handleRemoteOverrideCancel()
+        case .bolusEntry(let bolusCommand):
+            try await self.serviceDelegate?.handleRemoteBolus(amountInUnits: bolusCommand.amountInUnits)
+        case .carbsEntry(let carbCommand):
+            try await self.serviceDelegate?.handleRemoteCarb(
+                amountInGrams: carbCommand.amountInGrams,
+                absorptionTime: carbCommand.absorptionTime,
+                foodType: carbCommand.foodType,
+                startDate: carbCommand.startDate
+            )
+        case .autobolus(let autobolusCommand):
+            try await self.serviceDelegate?.handleRemoteAutobolus(activate: autobolusCommand.active)
+        case .closedLoop(let closedLoopComand):
+            try await self.serviceDelegate?.handleRemoteClosedLoop(activate: closedLoopComand.active)
         }
     }
+    
+    func commandSourceV1(_: RemoteCommandSourceV1, uploadError error: Error, notification: [String: AnyObject]) async throws {
+        
+        guard let uploader = self.uploader else {throw NightscoutServiceError.missingCredentials}
+        var commandDescription = "Loop Remote Action Error"
+        if let remoteNotification = try? notification.toRemoteNotification() {
+            commandDescription = remoteNotification.toRemoteAction().description
+        }
+        
+        let notificationJSON = try JSONSerialization.data(withJSONObject: notification)
+        let notificationJSONString = String(data: notificationJSON, encoding: .utf8) ?? ""
+        
```

Comment:
The JSON is included in the note which may not be desired. I added it just in case we want to use that JSON in Caregiver for the interim (rather than users needing to look in NS), while the Remote 2.0 pieces are being worked out.